packages/xchain-client/src/BaseXChainClient.ts

```diff
@@ -42,6 +43,7 @@ export abstract class BaseXChainClient implements XChainClient {
     this.chain = chain
     this.network = params.network || Network.Testnet
     this.feeBounds = params.feeBounds || { lower: 1, upper: Infinity }
+    this.customRequestHeaders = params.customRequestHeaders ?? { 'x-client-id': 'xchainjs-client' }
```

Comment:
Don't set a default value - see https://github.com/xchainjs/xchainjs-lib/pull/689#discussion_r1032818390

Reply:
If we dont set the default value, then if a user of an xchain-* lib does not send in a header value, all requests will fail no?  don't we want a "default"  value so that someone can pull our libs and immediately make requests?

Reply:
if a user of the lib wants to  turn off the default headers they would need to actively set customRequestHeaders = {}. i think this makes the most sense for onboarding, thoughts?

Reply:
> If we dont set the default value, then if a user of an xchain-* lib does not send in customRequestHeaders, all requests will fail no? don't we want a "default" value so that someone can pull our libs and immediately make requests?

From #https://github.com/xchainjs/xchainjs-lib/pull/689#discussion_r1032818390:
"Other servers than ninerealms.com might not accept ``'x-client-id'` in headers and can return an CORS error instead."

Reply:
> Yes. Just don't set a header in this case to other api endpoints than 9R (see prev. comment [about CORS]

So in this case we do something like this, `this.customRequestHeaders = params.customRequestHeaders ?? { '': '' }`
And in those Clients that use 9R  api endpoints we set the headers to default to this ` { 'x-client-id': 'xchainjs-client' }` ?

Reply:
@veado  have a look a the current impl, i think it does what needs to be done. however, we realized cosmos was missed (when used for making TC requests) and seems like the lib we use doesn't allow any addtional headers to be sent....i created an issue https://github.com/cosmos-client/cosmos-client-ts/issues/105 but I may need to patch the lib or find some other kind of workaround...

Reply:
> @veado have a look a the current impl, i think it does what needs to be done. however, we realized cosmos was missed (when used for making TC requests) and seems like the lib we use doesn't allow any addtional headers to be sent....i created an issue [cosmos-client/cosmos-client-ts#105](https://github.com/cosmos-client/cosmos-client-ts/issues/105) but I may need to patch the lib or find some other kind of workaround...

Yes, let's try to patch/fix it in a custom fork of `cosmos-client-ts` and use it as `dependency` (not `peerDependency`) in `xchain-*`. Open a PR with this patch, so we can use original `cosmos-client-ts` later again.

Reply:
the patch ended up being infeasible so we took the devs suggestion and used the default config, this is less than ideal, but wanted to get support for this completed and we can revisit if anyone reports an issue